src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,7 +53,51 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	/**
+	 * Checks that a string is either null, or has at least one non-whitespace
+	 * character and conforms to the specified regular expression.
+	 *
+	 * @param stringToCheck the string to check.
+	 * @param pattern       the pattern to validate against.
+	 * @param name          the name of the string to use in any error messages.
+	 * @param optional      whether or not the field is optional. If true, null is a valid value for the field.
+	 * @return the trimmed field.
+	 */
+	static String checkAgainstRegex(final String stringToCheck, final Pattern pattern, final String name, final boolean optional)
+			throws IllegalArgumentException {
+		final String checkedString = checkString(stringToCheck, name, optional);
+		if (checkedString == null) {
+			return null;
+		}
+		final Matcher m = pattern.matcher(checkedString);
+		if (m.find()) {
+			return checkedString;
+		}
+		throw new IllegalArgumentException(String.format(
+				"Illegal format for %s: \"%s\"\nIt should match the pattern \"%s\"",
+				name, stringToCheck, pattern.toString()));
+	}
```

Comment:
New method. There will be some more exciting patterns used in the next PR.

Reply:
Why not return the matcher so the calling method can get the matches and not have to rematch?

Reply:
Or return a value class with the checked string and the matcher